core/src/main/clojure/xtdb/buffer_pool.clj

```diff
@@ -436,12 +445,42 @@
 
 (set! *unchecked-math* :warn-on-boxed)
 
+(defn ->caffeine-cache ^com.github.benmanes.caffeine.cache.Cache [size ^Path local-disk-cache]
+  (log/infof "Creating local-disk-cache with size limit of %s bytes" size)
+  (let [cache (-> (Caffeine/newBuilder)
+                  (.maximumWeight size)
+                  (.weigher (reify Weigher
+                              (weigh [_ k v] 
```

Comment:
Our `pinning` mechanism in action - setting anything with a `:pinned?` value to weight 0 (so it will not be considered for weight connected eviction)